### PR TITLE
Enable building embedded distributions from standalone modules

### PR DESCRIPTION
**Pull request by labkey-tchad**
#### Rationale
In order to build embedded distributions from standalone modules, they need to be able to pull the embedded jar from Artifactory instead of building it from `:server:embedded`. This adds an `embedded` configuration to the distribution plugin that references ':server:embedded:jar' or a published embedded jar as needed.

#### Related Pull Requests
* https://github.com/LabKey/server/pull/99

#### Changes
* Enable building embedded distributions from standalone modules
* Exclude standard upgrade script and README from embedded distributions
* [Issue 43356: manual-upgrade.sh script fails to upgrade standalone installer](https://www.labkey.org/home/Developer/issues/Secure/issues-details.view?issueId=43356)

### CODE REVIEW

File: src/main/groovy/org/labkey/gradle/plugin/Distribution.groovy [hunk 1]
```diff
@@ -75,6 +75,12 @@ class Distribution implements Plugin<Project>
                 }
         project.configurations.distribution.setDescription("Artifacts of creating a LabKey distribution (aka installer)")
 
+        project.configurations
+                {
+                    embedded
+                }
+        project.configurations.distribution.setDescription("Artifacts for creating an embedded LabKey distribution (aka installer)")
```
Comment (labkey-susanh) [inline src/main/groovy/org/labkey/gradle/plugin/Distribution.groovy]:
```suggestion
        project.configurations.embedded.setDescription("Artifacts for creating an LabKey distribution (aka installer) with Tomcat embedded in it")
```

File: src/main/groovy/org/labkey/gradle/task/StopTomcat.groovy [hunk 2]
```diff
@@ -29,7 +29,7 @@ class StopTomcat extends DefaultTask
     @TaskAction
     void action()
     {
-        if (BuildUtils.useEmbeddedTomcat(project))
+        if (BuildUtils.useLocalEmbeddedTomcat(project))
```
Comment (labkey-susanh) [inline src/main/groovy/org/labkey/gradle/task/StopTomcat.groovy]: Someone who has built a standalone distribution that includes the embedded jar and then uses the stopTomcat task will have problems here.  Do we have any documentation or help for this situation?

File: src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy [hunk 3]
```diff
@@ -474,9 +470,9 @@ class ModuleDistribution extends DefaultTask
 
         ant.zip(destfile: getEmbeddedZipArchivePath()) {
             zipfileset(dir: project.buildDir, prefix: archiveName) { include(name: serverJarFile.getName()) }
-            zipfileset(dir: utilsDir.path, prefix: "${archiveName}/bin")
-            zipfileset(dir: project.buildDir, prefix: archiveName, filemode: 744){
-                include(name: "manual-upgrade.sh")
+
+            if (!simpleDistribution) {
+                zipfileset(dir: utilsDir.path, prefix: "${archiveName}/bin")
```
Comment (labkey-susanh) [inline src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy:478]: Are you sure you can get rid of the setting of the filemode for `manual-upgradle.sh`?  Are we no longer building installers from a Windows machine?  I think that's where this need arose.

Comment (labkey-tchad) [reply src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy:478]: This removes that script from embedded distributions altogether. I don't think it actually works for embedded distributions.

Comment (labkey-susanh) [reply src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy:478]: Got it.  Didn't see the larger context here.

**Review by labkey-susanh — APPROVED** [review]

**Pull request merged**